### PR TITLE
Correct type for request_token. Capture result of tx.Create

### PR DESCRIPTION
**Pull request by cgilmour**
Experiencing error when trying to get addresses from IPAM.
First one would succeed, and entry created in ipam_vms table.
After that, all fails, but doesn't accurately react to the error.

Error is: `2016/03/07 02:57:23 [Error 1062: Duplicate entry '' for key 'request_token']`
Instead of stopping, it'd proceed to allocate the same IP address for the host on multiple calls (so different pods/instances get the same address.
```
$ grep Constructing /var/log/upstart/romana-ipam.log 
2016/03/06 21:53:03 Constructing IP from Host IP 10.0.0.1/16, Tenant 2, Segment 1
2016/03/06 21:53:03 Constructing (167772160 << 24) | 167772160 | (2 << 12) | ( 1 << 8 ) | 3=10.0.33.3
2016/03/06 21:53:04 Constructing IP from Host IP 10.1.0.1/16, Tenant 2, Segment 1
2016/03/06 21:53:04 Constructing (167772160 << 24) | 167837696 | (2 << 12) | ( 1 << 8 ) | 3=10.1.33.3
2016/03/06 21:53:05 Constructing IP from Host IP 10.1.0.1/16, Tenant 2, Segment 1
2016/03/06 21:53:05 Constructing (167772160 << 24) | 167837696 | (2 << 12) | ( 1 << 8 ) | 3=10.1.33.3
```

This patch fixes both issues.
1) Replace request_token's type with NullString. It is tagged unique, but will default to an empty string (a non-NULL value from the DB's perspective).
2) When tx.Create() is called, it returns a *DB  object that contains the error. We need to keep this to check if an error occured.

### CODE REVIEW

File: ipam/store.go [hunk 1]
```diff
@@ -72,8 +72,8 @@ func (ipamStore *ipamStore) addVm(stride uint, vm *Vm) error {
 	vm.EffectiveSeq = effectiveVmSeq
 	ipamVm := IpamVm{Vm: *vm}
 	tx.NewRecord(ipamVm)
-	tx.Create(&ipamVm)
-	err := common.MakeMultiError(ipamStore.DbStore.Db.GetErrors())
+	tx = tx.Create(&ipamVm)
```
Comment (jbrendel) [inline ipam/store.go]: This line looks funny, but I don't know enough about Go database functions. Would probably be best to ensure that Greg also can have a look at it.

Comment (debedb) [inline ipam/store.go]: This line's cool in theory but I'm going to make a test. @cgilmour and I already connected on this.

Comment (jbrendel) [inline ipam/store.go]: Great, thank you!

**Comment by debedb** [comment]
Created https://paninetworks.kanbanize.com/ctrl_board/3/cards/278/details for future

**Comment by debedb** [comment]
ok looks good.

**Pull request merged**